fails to build with gcc-15: conflicting types for ‘finalize_arrays’

armci-mpi (0.4) is failing to build with gcc-15 (C23 standard):
```
mpicc.openmpi -DHAVE_CONFIG_H -I. -I./src  -I./src -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -Werror=implicit-function-declaration -ffile-prefix-map=/build/reproducible-path/armci-mpi-0.4=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection   -c -o tests/contrib/cg/cg.o tests/contrib/cg/cg.c
tests/contrib/cg/cg.c:146:6: error: conflicting types for ‘finalize_arrays’; have ‘void(int)’
  146 | void finalize_arrays(int dpc)
      |      ^~~~~~~~~~~~~~~
tests/contrib/cg/cg.c:25:6: note: previous declaration of ‘finalize_arrays’ with type ‘void(void)’
   25 | void finalize_arrays();
      |      ^~~~~~~~~~~~~~~
make[3]: *** [Makefile:1775: tests/contrib/cg/cg.o] Error 1
```
Full log at https://buildd.debian.org/status/fetch.php?pkg=armci-mpi&arch=amd64&ver=0.4-7&stamp=1761522189&raw=0

--- a/tests/contrib/cg/cg.c
+++ b/tests/contrib/cg/cg.c
@@ -22,7 +22,7 @@ static int niter;
 void read_and_create(int,char **);
 void computeminverser(double *,double *, double *);
 void computeminverse(double *,double *, int *,int *);
-void finalize_arrays();
+void finalize_arrays(int);
 extern void acg_matvecmul(double *,double *,double *,int *,int *);
 extern void acg_addvec(double *,double *,double *,double *, double *);
 extern void acg_2addvec(double *,double *, double *,double *, double *,

